cmd/ipfs-cluster-follow/main.go

```diff
@@ -306,6 +338,27 @@ func socketAddress(absPath, clusterName string) (multiaddr.Multiaddr, error) {
 	return ma, nil
 }
 
+func killFollower(absPath string) error {
+	conn, err := net.Dial("unix", filepath.Join(absPath, "api-socket"))
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	connFile, err := conn.(*net.UnixConn).File()
+	if err != nil {
+		return err
+	}
+	ucred, err := syscall.GetsockoptUcred(int(connFile.Fd()), syscall.SOL_SOCKET, syscall.SO_PEERCRED)
+	if err != nil {
+		return err
+	}
+	followerP, err := os.FindProcess(int(ucred.Pid))
+	if err != nil {
+		return err
+	}
+	return followerP.Signal(syscall.SIGTERM)
+}
```

Comment:
Assuming this indeed doesn't work on Windows, another option would be to have `ipfs-cluster-follow run` directly write its pid into a file, which can be read and terminated directly, without having to resort to socket tricks.

The pid-file creation code would be part of `cmdutils` to be able to re-use it. And ofc needs to be removed when the process dies.

Reply:
You're right, it's not work on Windows, and even we get the pid of ipfs-cluster-follow on Windows, maybe we also cannot gracefully terminate process on Windows. See 
- https://stackoverflow.com/questions/55092139/gracefully-terminate-a-process-on-windows/62733194#62733194
- https://github.com/golang/go/issues/6720#issuecomment-66087740
- https://github.com/iwdgo/sigintwindows

1. Windows cannot directly send SIGTERM signal to process like Unix-like Systems.
2. We can use `taskkill /PID <pid> /F` or below code to terminate process easily, but it will directly close the process and we cannot do anything before closing processes
```go
func KillFollower(pid int) error {
	handle, _ := syscall.OpenProcess(syscall.PROCESS_TERMINATE, false, uint32(pid))
	defer syscall.CloseHandle(handle)
	return syscall.TerminateProcess(handle, 0)
}
```
3. The one way to gracefully terminate process is use `GenerateConsoleCtrlEvent`, but we must start ipfs-cluster-follow with `REATE_NEW_PROCESS_GROUP` flag to get the pid of "console process group".
4. Maybe we can only add a self close function on `Client` interface bypassing limitations of the Windows.